### PR TITLE
Improve infra/ci

### LINKED ISSUE #3: Do we need Discussions?
We have it in the spec repo and many other languages SIGs.

e.g. https://github.com/open-telemetry/opentelemetry-specification/discussions

### PR DESCRIPTION
**Pull request by reyang**
Fixes #3.

## Changes

* Updated the Discussions link. @open-telemetry/demo-webstore-maintainers need to decide if we want to use Discussions or something else (Issues, Slack, etc.).
* Fixed the CI.
* Minor fixes (wording, links).

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/question.md
Comment (reyang) [inline .github/ISSUE_TEMPLATE/question.md:9]: @open-telemetry/demo-webstore-maintainers please advise if we should create/use Discussions or something else.

**Review by cartersocha — APPROVED** [review]

File: .github/ISSUE_TEMPLATE/question.md
Comment (cartersocha) [reply .github/ISSUE_TEMPLATE/question.md:9]: I think we should use / create it

Comment (reyang) [reply .github/ISSUE_TEMPLATE/question.md:9]: Okay, I'll merge this PR to unblock the CI. Then you can help to make these CI `required`.

We can discussion more in the upcoming SIG meeting if folks have other ideas.

**Pull request merged**
